### ISSUE TITLE
`block-scoped-var` throws an error if missing parameter in destructuring

### ISSUE THREAD

**Issue by dahjelle** [eslint/eslint#2111]
Running `eslint` on the following file:

```javascript
var foo = function([, bar]) {
  return bar;
};
```

throws this error

```
.../eslint/lib/rules/block-scoped-var.js:142
                    declarations.push(element.name);
                                             ^
TypeError: Cannot read property 'name' of null
    at .../eslint/lib/rules/block-scoped-var.js:142:46
    at Array.forEach (native)
    at declareByNodeType (.../eslint/lib/rules/block-scoped-var.js:141:31)
    at .../eslint/lib/rules/block-scoped-var.js:157:13
    at Array.forEach (native)
    at EventEmitter.functionHandler (.../eslint/lib/rules/block-scoped-var.js:156:21)
    at EventEmitter.emit (events.js:117:20)
    at Controller.controller.traverse.enter (.../eslint/lib/eslint.js:735:25)
    at Controller.__execute (.../eslint/node_modules/estraverse/estraverse.js:393:31)
    at Controller.traverse (.../eslint/node_modules/estraverse/estraverse.js:491:28)
```

I'm running `eslint` v0.17.1 with the following `.eslintrc` (with `strict` and `no-unused-vars` turned off for clarity)

```json
{
  "env": {
    "es6": true
  },
  "rules": {
    "strict": 0,
    "no-unused-vars": 0,
    "block-scoped-var": 2
  }
}
```

I imagine it should show a lint error (or perhaps another rule should), but I don't think it should crash.

**Comment by nzakas** [comment]
Thanks, we will get that fixed.

**Comment by dahjelle** [comment]
Awesome—thanks!

**Comment by nzakas** [comment]
Working on this.
